packages/shared/components/inputs/AssetAmountInput2.svelte

```diff
@@ -0,0 +1,105 @@
+    import { getMaxDecimalsFromTokenMetadata } from '@core/token/utils'
+    import {
+        IAsset,
+        TokenStandard,
+        convertToRawAmount,
+        formatTokenAmountDefault,
+        visibleSelectedAccountAssets,
+    } from '@core/wallet'
+    import Big from 'big.js'
+    import { AmountInput, FontWeight, InputContainer, Text } from '@ui'
+    import { getMarketAmountFromAssetValue } from '@core/market/utils'
+
+    export let disabled = false
+    export let asset: IAsset | undefined = $visibleSelectedAccountAssets?.baseCoin
+    export let rawAmount: string | undefined = undefined
+    export let unit: string | undefined = undefined
+    export let amount: string | undefined =
+        rawAmount && asset?.metadata
+            ? formatTokenAmountDefault(Number(rawAmount), asset.metadata, unit, false)
+            : undefined
+
+    let amountInputElement: HTMLInputElement
+    let error: string
+
+    $: amount, (error = '')
+    $: allowedDecimals = asset?.metadata && unit ? getMaxDecimalsFromTokenMetadata(asset.metadata, unit) : 0
+    $: availableBalance = asset?.balance?.available ?? 0
+    $: bigAmount = amount && asset?.metadata ? convertToRawAmount(amount, asset.metadata, unit) : 0
+    $: marketAmount = asset ? getMarketAmountFromAssetValue(bigAmount, asset) : undefined
+    $: rawAmount = bigAmount.toString()
+
+    export function validate(allowZeroOrNull = false): Promise<void> {
+        if (amount === undefined || asset?.metadata === undefined) {
+            return Promise.reject()
+        }
+        const amountAsFloat = parseCurrency(amount)
+        const isAmountZeroOrNull = !Number(amountAsFloat)
+        const requiresRawAmount =
+            (asset.metadata.standard === TokenStandard.BaseToken && unit === asset.metadata.subunit) ||
+            asset.metadata.decimals === 0
+        const bigAmount = convertToRawAmount(amount, asset.metadata, unit)
+
+        // Zero value transactions can still contain metadata/tags
+        error = ''
+        if (allowZeroOrNull && isAmountZeroOrNull) {
+            rawAmount = Big(0).toString()
+            return Promise.resolve()
+        } else if (isAmountZeroOrNull) {
+            error = localize('error.send.amountInvalidFormat')
+        } else if (requiresRawAmount && Number.parseInt(amount, 10).toString() !== amount) {
+            error = localize('error.send.amountNoFloat')
+        } else if (bigAmount.gt(Big(availableBalance))) {
+            error = localize('error.send.amountTooHigh')
+        } else if (bigAmount.lte(Big(0))) {
+            error = localize('error.send.amountZero')
+        } else if (!bigAmount.mod(1).eq(Big(0))) {
+            error = localize('error.send.amountSmallerThanSubunit')
+        }
+
+        if (error) {
+            return Promise.reject(error)
+        }
+        rawAmount = bigAmount.toString()
+        return Promise.resolve()
+    }
+</script>
+
+<!-- svelte-ignore a11y-click-events-have-key-events -->
```

Comment:
perhaps out of scope of this pr, but some ux improvements:
- the amount goes off the page when its too long, the size of the font shoul be proportional to the length
- if i type a decimal as the first value it should add a 0 to the string at the front
- should we even allow them to type more than the max?